controllers/drplacementcontrol_controller.go

```diff
@@ -962,7 +997,42 @@ func (d *DRPCInstance) runInitialDeployment() (bool, error) {
 
 	const done = true
 
-	// 1. Check if the user wants to use the preferredCluster
+	homeCluster, homeClusterNamespace := d.getHomeCluster()
+
+	if homeCluster == "" {
+		err := fmt.Errorf("PreferredCluster not set and unable to find home cluster in DRPCPlacementRule (%v)",
+			d.drpcPlacementRule)
+		// needStatusUpdate is not set. Still better to capture the event to report later
+		rmnutil.ReportIfNotPresent(d.reconciler.eventRecorder, d.instance, corev1.EventTypeWarning,
+			rmnutil.EventReasonDeployFail, err.Error())
+
+		return !done, err
+	}
+
+	d.log.Info(fmt.Sprintf("Using homeCluster %s for initial deployment, uPlRule Decision %v",
+		homeCluster, d.userPlacementRule.Status.Decisions))
+
+	clusterName, deployed := d.isDeployed(homeCluster)
+	if deployed && clusterName != homeCluster {
```

Comment:
Should this be `clusterName == homeCluster`?

Reply:
So if we detect that the app is already deployed somewhere else (not the homeCluster), then we are done.  If it is deployed to this homeCluster, then we need to proceed. The homeCluster is the configured preferredCluster or drpcPlacement generated cluster.